### PR TITLE
[kubernetes-sigs/secrets-store-csi-driver#1091]add optional presubmit for alibabacloud provider with secrets-store-csi

### PR DESCRIPTION
**Pull request by DahuK**
PR adds:
1. optional presubmit testing for alibabacloud provider with secrets-store-csi
2. external secret in order to use it in alibabacloud provider tests

### CODE REVIEW

**Comment by chases2** [comment]
Please get approval and review from someone in [config/jobs/kubernetes-sigs/secrets-store-csi-driver/OWNERS](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/secrets-store-csi-driver/OWNERS) who has context. If after that you still need further approval, feel free to assign to the prow oncall (or to me)

**Comment by DahuK** [comment]
@aramase @hixichen @ritazh
this has held for a long time, could you please take a time for reviewing this and help to move forward, thanks!

**Review by aramase — CHANGES_REQUESTED** [review]

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml
Comment (DahuK) [reply config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml]: > fix this, thanks!

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml
Comment (DahuK) [reply config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml]: updated

File: config/prow/cluster/build/build_kubernetes-external-secrets_customresource.yaml
Comment (DahuK) [reply config/prow/cluster/build/build_kubernetes-external-secrets_customresource.yaml:23]: yes, it looks like the project using archived [kubenrtes-external-secrets](https://github.com/external-secrets/kubernetes-external-secrets#alibaba-cloud-kms-secret-manager), and I have tested the alicloudSecretsManager backend in our cluster

**Comment by DahuK** [comment]
@aramase  my apologies for the delayed update. Please review it and another relevant [PR](https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/1091),   and let me know if there's anything I need to add.

**Comment by BenTheElder** [comment]
Relevant announcements:

https://groups.google.com/a/kubernetes.io/g/dev/c/p6PAML90ZOU/m/11sDguoxAQAJ
https://groups.google.com/a/kubernetes.io/g/dev/c/qzNYpcN5la4

We're not permitting any more external secrets like this, which in retrospect were a HUGE mistake.

If you want to run on Kubernetes's community CI, you need to provide resources to SIG K8s Infra via an agreement with the CNCF so we understand:
- how much is available
- how long are the resources committed
- visibility into usage (so we know when we're about to run out)
- access granted to the CNCF, and then SIG K8s Infra leads, and then also some community members, so we're not left with defunct resources nobody can control when employees that set them up leave

IF they meet those requirements, we can arrange to use them from the community clusters.

`cluster: default` which has these external secrets is another old company-internal resource that is being shut down.

Closing this as per https://github.com/kubernetes/test-infra/issues/33226

**Pull request closed without merging**
